src/clipboard.rs

```diff
@@ -0,0 +1,46 @@
+use raw_window_handle::RawDisplayHandle;
+
+#[cfg(not(any(target_os = "macos", windows)))]
+use copypasta::{
+    wayland_clipboard,
+    x11_clipboard::{Primary as X11SelectionClipboard, X11ClipboardContext},
+};
+
+use copypasta::{ClipboardContext, ClipboardProvider};
+
+pub struct Clipboard {
+    pub clipboard: Box<dyn ClipboardProvider>,
+    pub selection: Option<Box<dyn ClipboardProvider>>,
+}
+
+impl Clipboard {
+    /// # Safety
+    /// The `display` must be valid as long as the returned Clipboard exists.
+    pub unsafe fn new(
+        #[allow(unused_variables)] /* on some platforms */ display: RawDisplayHandle,
+    ) -> Self {
+        #[cfg(not(any(target_os = "macos", windows)))]
+        if let RawDisplayHandle::Wayland(display) = display {
+            let (selection, clipboard) =
+                wayland_clipboard::create_clipboards_from_external(display.display);
+            return Self {
+                clipboard: Box::new(clipboard),
+                selection: Some(Box::new(selection)),
+            };
+        }
+
+        #[cfg(not(any(target_os = "macos", windows)))]
+        return Self {
+            clipboard: Box::new(ClipboardContext::new().unwrap()),
+            selection: Some(Box::new(
+                X11ClipboardContext::<X11SelectionClipboard>::new().unwrap(),
+            )),
+        };
+
+        #[cfg(any(target_os = "macos", windows))]
+        return Self {
+            clipboard: Box::new(ClipboardContext::new().unwrap()),
+            selection: None,
+        };
+    }
```

Comment:
maybe add `get_contents` and `set_contents` to `Clipboard` 

so that user can just call `Clipboard::get_contents()` `Clipboard::set_contents(contents)` without needs to import `CLIPBOARD` and lock it.